### PR TITLE
support custom network id

### PR DESCRIPTION
**Pull request by yosriz**
* change network to String for supporting custom network passphrase
* change kin asset creation to be done lazily, to properly be able to implement custom asset
 (constructors must not invoke overridable methods)

### CODE REVIEW

File: dependencies.gradle [hunk 1]
```diff
@@ -9,7 +9,7 @@ ext {
     mockWebServerVersion = '3.8.1'
     robolectricVersion = '3.6.1'
     hamcrestVersion = '1.3'
-    stellarSDKVersion = '3b05787858d847be59f3dc6333ed44887ce5cc1b'
+    stellarSDKVersion = 'bfc53061e14ff6d2b83b44f5adba5ce1d1a91466'
```
Comment (OrenZak) [inline dependencies.gradle:12]: Can't we use versions?

File: kin-core/src/main/java/kin/core/ServiceProvider.java [hunk 2]
```diff
@@ -73,18 +63,25 @@ final public String getProviderUrl() {
...
     }
 
     final public boolean isMainNet() {
-        return networkId == NETWORK_ID_MAIN;
+        return NETWORK_ID_MAIN.equals(network.getNetworkPassphrase());
     }
 
     final KinAsset getKinAsset() {
+        if (kinAsset == null) {
+            kinAsset = new KinAsset(getAssetCode(), getIssuerAccountId());
```
Comment (OrenZak) [inline kin-core/src/main/java/kin/core/ServiceProvider.java:76]: Are we not afraid of multithreading?

**Review by OrenZak — APPROVED** [review]
👍 
just see my comments

File: dependencies.gradle (hunk 1 above)
Comment (yosriz) [reply dependencies.gradle:12]: I did in the past, but I don't want to maintain versions for that, as this is a internal library (meaning we are not targeting it for outside use currently), I would prefer actually to have `android-stellar-sdk` as submoudle, but didn't have the chance to do it by now

File: kin-core/src/main/java/kin/core/ServiceProvider.java (hunk 2 above)
Comment (yosriz) [reply kin-core/src/main/java/kin/core/ServiceProvider.java:76]: Good question, didn't take it into consideration. 
I think our approach should be that `KinClient`/`KinAccount` not thread-safe by default, as I don't see a point for that, if there will be a demand, we should consider it in the future.

**Pull request merged**